node/packages/aws-lambda-sdk/instrument/index.js

```diff
@@ -98,15 +99,27 @@ const reportResponse = async (response, context, endTime) => {
   await sendTelemetry('request-response', payloadBuffer);
 };
 
+let requestCounter = 0;
+let lastCounterResetTime = 0;
+
 const reportTrace = ({ isErrorOutcome }) => {
   // Sample out 80% of traces in production mode if no warning or error event was reported
```

Comment:
I think I am following looking at the code but not 100% sure. I think an updated comment would help 🙏 

Does this sound right?

```
/**
 * This function determines if a trace should be sampled or not
 * 
 * 
 * We never sample traces when
 *    Trace contains Errors or Warnings
 *    Dev Mode = enabled
 *    Debug Mode = enabled
 *    Cold start
 * 
 * Given none of the conditions above are met we will sample out 90% of traces
*/
```

I think I am missing the part about `isApiEvent` 🤔

Reply:
Good point, I forgot to update this comment. I've updated it now